internal/providers/providers.go

```diff
@@ -78,30 +78,27 @@ func GetCredentialStateForProvider(
 	s db.Store,
 	cryptoEngine crypto.Engine,
 	provCfg *serverconfig.ProviderConfig,
-) string {
-	var credState string
+) (string, error) {
```

Comment:
This addresses a latent bug in this codepath found during testing. There are two places where this function gets called: one place where the error can be safely ignored, and one where it should not be ignored. Since error was always being swallowed here, it led to a situation where the second callsite would call this function, proceed when it should not, and end up with weird behaviour.

Refactored the function to return an error. As a nice side effect, the controlflow became far more simple.